tests/dummy/app/templates/docs/glimmer-components.md

```diff
@@ -0,0 +1,109 @@
+
+## A simple component
+
+A *very* simple Glimmer component which lets you change the count of a value might look like this:
+
+```htmlbars
+<button {{on "click" this.minus}}>&minus;</button>
+{{this.count}}
+<button {{on "click" this.plus}}>+</button>
+```
+
+```ts
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking'
+import { action } from '@ember/object';
+
+export default class Counter extends Component {
+  @tracked count = 0;
+  
+  @action
+  plus() {
+    this.count += 1;
+  }
+  
+  @action
+  minus() {
+    this.count -= 1;
+  }
+}
+```
+
+Notice that there are no type declarations here – but this *is* actually a well-typed component. The type of `count` is `number`, and if we accidentally wrote something like `this.count = null` the compiler would give us an error.
+
+## A component with arguments
+
+So far so good, but of course most components aren’t quite this simple! Instead, they’re invoked by other templates and they can invoke other components themselves in their own templates.
+
+Glimmer components can receive both *arguments* and *attributes* when they are invoked. When you are working with a component’s backing class, you have access to the arguments but *not* to the attributes. The arguments are passed to the constructor, and then available as `this.args` on the component instance afterward. Let’s imagine a component which just logs the names of its arguments when it is first constructed:
+
+```ts
+import Component from '@glimmer/component';
+
+export default class ArgsDisplay extends Component {
+  constructor(owner: unknown, args: {}) {
+    super(owner, args);
+
+    Object.keys(args).forEach(argName => {
+      console.log(argName);
+    });
+  }
+}
+```
+
+<aside>
+
+If you’re used to the classic Ember Object model, there are two important differences in the constructor itself:
+
+- we use `super` instead of `this._super`
+- we *must* call `super` before we do anything else
+
+</aside>
+
+Notice that we have to start by calling `super` with `owner` and `args`. This may be a bit different from what you’re used to in Ember or other frameworks, but is normal for sub-classes in TypeScript. If the compiler just accepted any `...arguments`, a lot of potentially *very* unsafe invocations would go through. So, instead of using `...arguments`, we explicitly pass the *specific* arguments and make sure their types match up with what the super-class expects.
+
+The types for `owner` here and `args` line up with what the `constructor` for Glimmer components expect. The `owner` is specified as `unknown` because this is a detail we explicitly *don’t* need to know about. The `args` are `object` because a Glimmer component *always* receives an object containing its arguments, even if the caller didn’t pass anything: then it would just be an empty object.
```

Comment:
> The `args` are `object`

for consistency, I would update the constructor signature to `args: object` to ensure readers aren't burdened with thinking about whether the empty interface type `{}` and the type `object` are equivalent (I myself had to check).

Reply:
I need to change the comment here to match the original, actually – I had updated the definition to match what the types on `Component` specify precisely to *avoid* explaining that. What this makes me realize is we probably need an `aside` linking to the explanation of the difference of the two.